app/controllers/admin/web_hooks_controller.rb

```diff
@@ -87,14 +87,17 @@ def destroy
   def list_events
     limit = 50
     offset = params[:offset].to_i
+    events = @web_hook.web_hook_events
+    if params[:status] == "successful"
+      events = events.successful
+    elsif params[:status] == "failed"
+      events = events.failed
+    end
+    events = events.limit(limit).offset(offset)
 
     json = {
-      web_hook_events:
-        serialize_data(
-          @web_hook.web_hook_events.limit(limit).offset(offset),
-          AdminWebHookEventSerializer,
-        ),
-      total_rows_web_hook_events: @web_hook.web_hook_events.count,
+      web_hook_events: serialize_data(events, AdminWebHookEventSerializer),
+      total_rows_web_hook_events: events.count,
```

Comment:
Infinite scrolling is not working because `events.count` is the number of events _we are currently sending_. So the frontend thinks all events are loaded even when there are more.

Before you set the limit and offset of `events` you can get a count like
```ruby
total = events.count
events = events.limit(limit).offset(offset)

json = {
      web_hook_events: serialize_data(events, AdminWebHookEventSerializer),
      total_rows_web_hook_events: total,
      # ....
```